ts/features/idpay/initiative/details/components/TimelineTransactionCards.tsx

```diff
@@ -1,109 +1,101 @@
-          transaction.operationDate
-        )}, ${getHourAndMinuteFromDate(
-          transaction.operationDate
-        )} · ${formatNumberAmount(Math.abs(transaction.amount), true)}`}
-      </LabelSmall>
-    </View>
-    <H4> {`${formatNumberAmount(transaction.amount, false)} €`}</H4>
-  </BaseTransactionCard>
-);
-
-export const OnboardingTransactionCard = ({
-  transaction
-}: TransactionProps<OnboardingOperationDTO>) => (
-  <BaseTransactionCard>
-    <Text>LOGO</Text>
-    <View hspacer />
-    <View style={IOStyles.flex}>
-      <H4>{transaction.operationType}</H4>
-      <LabelSmall weight="Regular" color="bluegrey">
-        {`${formatDateAsShortFormat(
-          transaction.operationDate
-        )}, ${getHourAndMinuteFromDate(transaction.operationDate)}`}
-      </LabelSmall>
-    </View>
-  </BaseTransactionCard>
-);
+export const renderTimelineOperationCard = (transaction: OperationListDTO) => {
+  const hasAmount = "amount" in transaction;
+  return (
+    <ListItem style={styles.spaceBetween}>
+      <View
+        style={[
+          IOStyles.flex,
+          IOStyles.row,
+          styles.alignCenter,
+          styles.sidePadding
+        ]}
+      >
+        {"brandLogo" in transaction ? (
+          <Image
+            style={styles.imageSize}
+            source={{ uri: transaction.brandLogo }}
+          />
+        ) : (
+          iconMap[transaction.operationType]
+        )}
+        <View hspacer />
+        <View style={IOStyles.flex}>
+          <H4>
+            {operationTypeMap[transaction.operationType](
+              "maskedPan" in transaction ? transaction.maskedPan : ""
+            )}
+          </H4>
+          <LabelSmall weight="Regular" color="bluegrey">
+            {`${formatDateAsShortFormat(
+              transaction.operationDate
+            )}, ${getHourAndMinuteFromDate(transaction.operationDate)} ${
+              hasAmount
+                ? "· " + formatNumberAmount(Math.abs(transaction.amount), true)
+                : ""
+            }`}
+          </LabelSmall>
+        </View>
+        {hasAmount ? (
+          <H4> {`${formatNumberAmount(transaction.amount, false)} €`}</H4>
+        ) : null}
+      </View>
+    </ListItem>
+  );
+};
 
-export const InstrumentOnboardingCard = ({
-  transaction
-}: TransactionProps<InstrumentOperationDTO>) => (
-  <BaseTransactionCard>
-    <Text>LOGO</Text>
-    <View hspacer />
-    <View style={IOStyles.flex}>
-      <H4>
-        {transaction.operationType} {transaction.maskedPan}
-      </H4>
-      <LabelSmall weight="Regular" color="bluegrey">
-        {`${formatDateAsShortFormat(
-          transaction.operationDate
-        )}, ${getHourAndMinuteFromDate(transaction.operationDate)}`}
-      </LabelSmall>
-    </View>
-  </BaseTransactionCard>
-);
+type TypesForIcons =
+  | IbanOperationTypeEnum
+  | OnboardingOperationTypeEnum
+  | RefundOperationTypeEnum;
+const iconMap: Record<TypesForIcons, React.ReactNode> = {
```

Comment:
What happens if this map is access by a key not present in `TypesForIcons`?
Wouldn't be safer to use a switch in a function (e.g. `renderOperationIcon`) inside the component that defaults to `null`?